modules/packages/ZMQ.chpl

```diff
@@ -890,8 +895,16 @@ module ZMQ {
         // conditionally have ZeroMQ free the memory.
         // 
         // Note: the string factory below can throw DecodeError
+        // Another note: The bytes copies are parallelized in this file
+        // only temporarily, where long term we would like to do that
+        // parallelization in the internal bytes module, but are having
+        // to delay that due to some resolution issues. Also, the string
+        // assignment can likely be parallelized similarly, but was not
+        // included in this temporary effort.
         var copy = if isString(T) then createStringWithNewBuffer(x=data)
-                                  else createBytesWithNewBuffer(x=data);
+                   else if data.size < parallelAssignThreshold then createBytesWithNewBuffer(x=data)
```

Comment:
I think it'll be a little cleaner to move this size check into `parallelCreateBytesWithNewBuffer`.